ncm-spma/src/main/pan/components/spma/functions.pan

```diff
@@ -101,8 +101,10 @@ function resolve_pkg_rep = {
     arg = repository list
 }
 function purge_rep_list = {
-    foreach (i;rep;SELF) {
-        rep['contents'] = null;
+    if (is_list(SELF)) {
```

Comment:
then do a `if defined(SELF)` instead of `is_list` (the current code will happily take string, dicts and what not and ignore it. i'd rather have it fail in those situations then just do nothing.